src/main/scala/scalaz/stream/io.scala

```diff
@@ -293,10 +297,12 @@ object io {
     @tailrec
     override def close() {
       if (cur.isHalt && chunks.isEmpty) {
-        chunks = null
+        chunks = Nil
```

Comment:
Much more reasonable. I had a specific idea behind `null`, but `Nil` serves the purpose much better.